@afni_refacer_run kabooms with "AFNI can't deal with 5 dimensional NIfTI" on a regular anatomical 3D NIfTI

```
(dev3) yoh@typhon:/mnt/DATA/data/yoh/1076_spacetop$ singularity exec -B /mnt/DATA/data/yoh/afni/src/scripts_install/@afni_refacer_run:/usr/bin/@afni_reacer_run:ro -B $PWD/../atlases:/usr/share/AFNI/atlases:ro -B $PWD:$PWD:ro -B $PWD/derivatives:$PWD/derivatives:rw -B $PWD/scratch:$PWD/scratch:rw -B $PWD/../dsst-defacing-pipeline/src:/opt/dsst-defacing-pipeline:ro --env AFNI_ATLAS_PATH=/usr/share/AFNI/atlases --scratch $PWD/scratch --pwd $PWD --net --network none ../dsst-defacing-pipeline/dsst-defacing-pipeline-test.sing @afni_refacer_run  -input sub-0001/ses-01/anat/sub-0001_ses-01_acq-MPRAGEXp3X08mm_T1w.nii.gz -mode_all -prefix derivatives/dss-defacing-2/outtry
++ Found input file:   sub-0001/ses-01/anat/sub-0001_ses-01_acq-MPRAGEXp3X08mm_T1w.nii.gz

++ AFNI refacer, version      = 2.4
++ AFNI template dataset      = /usr/share/AFNI/atlases/MNI152_2009_template_SSW.nii.gz
++ AFNI refacer shell dataset = /usr/share/AFNI/atlases/afni_refacer_shell_sym_1.0.nii.gz
++ Mode for running           = ALL

*+ WARNING:   If you are performing spatial transformations on an oblique dset,
  such as sub-0001/ses-01/anat/sub-0001_ses-01_acq-MPRAGEXp3X08mm_T1w.nii.gz,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running:
     3dWarp -deoblique
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:sub-0001/ses-01/anat/sub-0001_ses-01_acq-MPRAGEXp3X08mm_T1w.nii.gz is 22.056934 degrees from plumb.
++ 3dcopy: AFNI version=AFNI_99.99.99 (May 22 2024) [64-bit]
++ 3dUnifize: AFNI version=AFNI_99.99.99 (May 22 2024) [64-bit]
** AFNI can't deal with 5 dimensional NIfTI(/mnt/DATA/data/yoh/1076_spacetop/derivatives/dss-defacing-2/__work_refacer.outtry.Y6axhoizawI/tmp.00.INPUT.nii)
** FATAL ERROR: Can't open dataset 'tmp.00.INPUT.nii'
** Program compile date = May 22 2024
++ 3dAllineate: AFNI version=AFNI_99.99.99 (May 22 2024) [64-bit]
++ Authored by: Zhark the Registrator
** FATAL ERROR: can't open --source dataset 'tmp.01.uni.nii' :-(
...
```

although that image is just 3d:

```
(dev3) yoh@typhon:/mnt/DATA/data/yoh/1076_spacetop$ nib-ls /mnt/DATA/data/yoh/1076_spacetop/derivatives/dss-defacing-2/__work_refacer.outtry.Y6axhoizawI/tmp.00.INPUT.nii
/mnt/DATA/data/yoh/1076_spacetop/derivatives/dss-defacing-2/__work_refacer.outtry.Y6axhoizawI/tmp.00.INPUT.nii int16 [224, 320, 320] 0.80x0.80x0.80  #exts: 1 sform
```

```
(dev3) yoh@typhon:/mnt/DATA/data/yoh/1076_spacetop$ singularity exec -B /mnt/DATA/data/yoh/afni/src/scripts_install/@afni_refacer_run:/usr/bin/@afni_reacer_run:ro -B $PWD/../atlases:/usr/share/AFNI/atlases:ro -B $PWD:$PWD:ro -B $PWD/derivatives:$PWD/derivatives:rw -B $PWD/scratch:$PWD/scratch:rw -B $PWD/../dsst-defacing-pipeline/src:/opt/dsst-defacing-pipeline:ro --env AFNI_ATLAS_PATH=/usr/share/AFNI/atlases --scratch $PWD/scratch --pwd $PWD --net --network none ../dsst-defacing-pipeline/dsst-defacing-pipeline-test.sing 3dinfo /mnt/DATA/data/yoh/1076_spacetop/derivatives/dss-defacing-2/__work_refacer.outtry.Y6axhoizawI/tmp.00.INPUT.nii
++ 3dinfo: AFNI version=AFNI_99.99.99 (May 22 2024) [64-bit]

Dataset File:    /mnt/DATA/data/yoh/1076_spacetop/derivatives/dss-defacing-2/__work_refacer.outtry.Y6axhoizawI/tmp.00.INPUT.nii
Identifier Code: XYZ_AvQZBHMfret5Uv9n9gqa_g  Creation Date: Fri May 24 13:46:10 2024
Template Space:  ORIG
Dataset Type:    Anat Bucket (-abuc)
Byte Order:      LSB_FIRST {assumed} [this CPU native = LSB_FIRST]
Storage Mode:    NIFTI
Storage Space:   45,875,200 (46 million) bytes
Geometry String: "MATRIX(-0.799523,0.027649,0.001149,85.29972,-0.026071,-0.741449,-0.299289,147.848,-0.009279,-0.299147,0.741906,-97.50206):224,320,320"
Data Axes Tilt:  Oblique (22.057 deg. from plumb)
Data Axes Approximate Orientation:
  first  (x) = Left-to-Right
  second (y) = Posterior-to-Anterior
  third  (z) = Inferior-to-Superior   [-orient LPI]
R-to-L extent:   -93.101 [R] -to-    85.300 [L] -step-     0.800 mm [224 voxels]
A-to-P extent:  -107.352 [A] -to-   147.848 [P] -step-     0.800 mm [320 voxels]
I-to-S extent:   -97.502 [I] -to-   157.698 [S] -step-     0.800 mm [320 voxels]
Number of values stored at each pixel = 1
  -- At sub-brick #0 '?' datum type is short:            0 to           573

----- HISTORY -----
[yoh@typhon: Fri May 24 13:46:10 2024] {AFNI_99.99.99:Linux_cmake} 3dcopy sub-0001/ses-01/anat/sub-0001_ses-01_acq-MPRAGEXp3X08mm_T1w.nii.gz derivatives/dss-defacing-2/__work_refacer.outtry.Y6axhoizawI/tmp.00.INPUT.nii
```
```

note that circle ci still runs all the testing and manages to build/upload docker images but does fail testing, e.g. https://app.circleci.com/pipelines/github/afni/afni/5870/workflows/c7035163-5af3-4dd8-b3a6-e14fc4213df9 has
![image](https://github.com/afni/afni/assets/39889/95c0e103-310d-496f-a926-c2d00567a61d)

with

```
=================================== FAILURES ===================================

___________________________ scripts/test_3dTstat.py ____________________________

[gw0] linux -- Python 3.6.9 /usr/bin/python3

worker 'gw0' crashed while running 'scripts/test_3dTstat.py::test_3dTstat_basic[cvarinv]'

=============================== warnings summary ===============================

../../../user_pip_packages/lib/python3.6/site-packages/requests/__init__.py:104

../../../user_pip_packages/lib/python3.6/site-packages/requests/__init__.py:104

../../../user_pip_packages/lib/python3.6/site-packages/requests/__init__.py:104

  /opt/user_pip_packages/lib/python3.6/site-packages/requests/__init__.py:104: RequestsDependencyWarning: urllib3 (1.26.18) or chardet (5.0.0)/charset_normalizer (2.0.12) doesn't match a supported version!

    RequestsDependencyWarning)



scripts/test_3dTstat.py::test_3dTstat_basic[cvarinv]

scripts/test_3dTstat.py::test_3dTstat_basic[cvarinv]

  /opt/****/src/tests/****_test_utils/data_management.py:355: DeprecationWarning: use poll_interval instead of poll_intervall

    dl_lock.acquire(poll_intervall=1)



-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

=========================== short test summary info ============================

FAILED scripts/test_3dTstat.py::test_3dTstat_basic[cvarinv]

SKIPPED [1] scripts/test_3dAllineate.py:17: reason=The comparison of to affines should use the expected distance of movement for a point drawn from a 3D cloud.

! xdist.dsession.Interrupted: <WorkerController gw0> received keyboard-interrupt !

============= 1 failed, 2 passed, 1 skipped, 5 warnings in 36.25s ==============


Exited with code exit status 2
```

so I guess no specific test for our problem at hands...

--- a/src/nifti/nifticdf/CMakeLists.txt
+++ b/src/nifti/nifticdf/CMakeLists.txt
@@ -8,7 +8,7 @@ set_target_properties(
     "${CMAKE_CURRENT_LIST_DIR}/nifticdf.h"
   )
 target_compile_options(${NIFTI_CDFLIB_NAME} PRIVATE "-D__COMPILE_UNUSED_FUNCTIONS__")
-target_link_libraries(${NIFTI_CDFLIB_NAME} PUBLIC ${NIFTI_PACKAGE_PREFIX}niftiio)
+target_link_libraries(${NIFTI_CDFLIB_NAME} PRIVATE ${NIFTI_PACKAGE_PREFIX}niftiio)
 get_lib_version_vars("nifticdf_version.h" NIFTICDF_VERSION NIFTICDF_MAJOR_VERSION)
 if(BUILD_SHARED_LIBS)
     set_target_properties(${NIFTI_CDFLIB_NAME}

